Fixes and enhancements for the Tahoma platform

glpatcern opened the pull request:
## Description:

Hi there,

Related to PR #11538, here you find a few more fixes and enhancements. They all work on my installation, will follow up for the automated linting/CI tests.

A breaking change is commit 963b34e that changes the entity IDs in HA. For this, I appreciate some feedback from @philklei as the original author and @bakedraccoon as the contributor of the above-mentioned PR.

**Related issue (if applicable):**

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):**

## Example entry for `configuration.yaml` (if applicable):

## Checklist:
  - [x] The code change is tested and works locally.

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [ ] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [ ] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

glpatcern committed:
Strip off the RTS/IO ID from the entity ID

glpatcern committed:
Ignore exception thrown when the device does not provide an active state

glpatcern committed:
Send actions with a label for easier identification in the Tahoma log

glpatcern committed:
Linting

bakedraccoon commented:
I think stripping off the entity id is a good choice. It's currently incredibly messy since it adds 24+ numbers that are already in 'tahoma_device_id' 👍

pvizeli commented on homeassistant/components/cover/tahoma.py:
```diff
@@ -33,7 +33,13 @@ class TahomaCover(TahomaDevice, CoverDevice):
+            eid = eid[:eid.rfind('_rts')]
+        elif eid.rfind('_io') > 0:
+            eid = eid[:eid.rfind('_io')]
+        self.entity_id = eid
```
Do you realy need to set the entity_id and don't let's auto generate from internal core/name ?

glpatcern replied:
The point is that - as mentioned above - the current entity_id looks like ```<custom_name>_rts_<NNN>```, where ```<NNN>``` is a very long numeric ID internally used by the TaHoma box to reference the shutters. This makes writing automations pretty unpractical (and the produced code unreadable).

IMHO Home Assistant should abstract from such platform-dependent details and expose high-level identifiers, so I believe it is definitely worth to override the auto-generated identifiers here.

bakedraccoon replied:
@pvizeli Currently an entity id look like this: ```cover.bedroom_middle_io12345678901234567890``` . Having ```cover.bedroom_middle``` would be more desireable, right?

pvizeli replied:
Yes. Home-Assistant generate automatic entity_id with name. Remove the entity_id and the core do that automatic. For unique_id, you can only use the `NN` if they is unique.

glpatcern replied:
So the question is about the pre-existing statement at [L36](https://github.com/home-assistant/home-assistant/pull/11547/files/5a86b1986610cf02d45f09f9d11ee95e2c69d4db#diff-c65eb52376398392f1395f8127364078L36), right? Do I understand correctly that leaving entity_id unassigned will make Home Assistant auto-generate an appropriate name?
There's no question about ```unique_id``` including the full identifier.

glpatcern replied:
Answering to my own question: yes. Even better, the patch will be to remove L36 altogether.

glpatcern committed:
Strip off the RTS/IO ID from the entity ID, take 2

As per suggestions, let HA do the standard initialization and assign
an appropriate entity_id, instead of overriding it with the lengthy
unique_id coming from the TaHoma devices.

glpatcern commented:
This should now be all good, the travis CI failed only because of a timeout.

bakedraccoon commented:
You're importing ```ENTITY_ID_FORMAT``` but it's not being used. https://github.com/glpatcern/home-assistant/blob/a061418e9e9037ec2af1004f26d56a7ffb2ed625/homeassistant/components/cover/tahoma.py#L10

glpatcern commented:
Good point, sorry to have overlooked it. Being at it, I completely removed the constructor and extended the commit message to explain why we just want to use the default behavior from HA.

glpatcern commented:
And yes, this time no timeouts and no warnings, so should be good to go.
